src/Microsoft.Identity.Client/Exceptions/MsalExceptionFactory.cs

```diff
@@ -52,66 +53,53 @@ public static Exception GetClientException(string errorCode, string errorMessage
         }
 
         /// <summary>
-        ///     Throws an <see cref="MsalClientException" /> exception
-        /// </summary>
-        /// <param name="errorCode">The error code</param>
-        /// <param name="errorMessage">A user friendly message</param>
-        /// <param name="httpResponse"></param>
-        public static Exception GetServiceException(string errorCode, string errorMessage, IHttpWebResponse httpResponse)
-        {
-            ValidateRequiredArgs(errorCode, errorMessage);
-            return GetServiceException(errorCode, errorMessage, null, ExceptionDetail.FromHttpResponse(httpResponse));
-        }
-
-        /// <summary>
-        ///     Throws an <see cref="MsalClientException" /> exception
-        /// </summary>
-        /// <param name="errorCode">The error code</param>
-        /// <param name="errorMessage">A user friendly message</param>
-        /// <param name="exceptionDetail">More exception params</param>
-        public static Exception GetServiceException(string errorCode, string errorMessage, ExceptionDetail exceptionDetail)
```

Comment:
As part of this change I did 2 minor refactoriongs: got rid of ExceptionDetail and of IHttpWebResponse. They existed to bridge the gap between similar ADAL and MSAL concepts, but since we separated the repos, this is no longer a concern and just add clutter.

The MsalExceptionFactory is now able to create exceptions based on HttpResponse and OAuth2Response objects directly, not needing an intermediary ExceptionDetail object.

Reply:
nice!

---
In reply to: [251456229](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/841#discussion_r251456229) [](ancestors = 251456229)